File: C/heapsort/deepseek-r1-distill-llama-70b/deepseek-r1-distill-llama-70b.c
#include <stdlib.h>
#include <time.h>

void sift_down(int *array, int size, int index) {
    int left = 2 * index + 1;
    int right = 2 * index + 2;
    int largest = index;

    if (left < size && array[left] > array[largest])
        largest = left;

    if (right < size && array[right] > array[largest])
        largest = right;

    if (largest != index) {
        int temp = array[index];
        array[index] = array[largest];
        array[largest] = temp;
        sift_down(array, size, largest);
    }
}

void heap_sort(int *array, int size) {
    // Build the heap
    for (int i = size / 2 - 1; i >= 0; --i)
        sift_down(array, size, i);

    // Extract elements one by one
    for (int i = size - 1; i >= 0; --i) {
        int temp = array[0];
        array[0] = array[i];
        array[i] = temp;
        sift_down(array, i, 0);
    }
}

int main() {
    srand(time(NULL));
    int size = 1000;
    int *array = (int *)malloc(size * sizeof(int));

    for (int i = 0; i < size; ++i)
        array[i] = rand() % 10000 + 1;

    heap_sort(array, size);

    free(array);
    return 0;
}

File: Python/heapsort/deepseek-r1-distill-llama-70b/deepseek-r1-distill-llama-70b.py
def heapify(arr, n, i):
    """
    Transforms a subtree rooted at index i into a max-heap.
    
    :param arr: The array to heapify.
    :param n: The size of the heap.
    :param i: The index of the root of the subtree.
    :return: None
    """
    largest = i
    left = 2 * i + 1
    right = 2 * i + 2

    if left < n and arr[left] > arr[largest]:
        largest = left

    if right < n and arr[right] > arr[largest]:
        largest = right

    if largest != i:
        arr[i], arr[largest] = arr[largest], arr[i]
        heapify(arr, n, largest)


def heap_sort(arr):
    """
    Sorts an array using the Heap Sort algorithm.
    
    :param arr: The array to be sorted.
    :return: The sorted array in ascending order.
    """
    n = len(arr)

    # Build a max-heap
    for i in range(n // 2 - 1, -1, -1):
        heapify(arr, n, i)

    # Extract elements one by one
    for i in range(n - 1, 0, -1):
        arr[0], arr[i] = arr[i], arr[0]
        heapify(arr, i, 0)

    return arr


import random

def main():
    arr = [random.randint(0, 10000) for _ in range(1000)]
    heap_sort(arr)

if __name__ == "__main__":
    main()